### ISSUE TITLE
[Doc]: Garbled menu widget example output 

### ISSUE THREAD

**Issue by StefRe** [matplotlib/matplotlib#27358]
### Documentation Link

https://matplotlib.org/devdocs/gallery/widgets/menu.html

### Problem

![grafik](https://github.com/matplotlib/matplotlib/assets/19879328/d2d021ad-fa03-4fb4-adfd-9bcf8b75b3af)

### Suggested improvement

Should be as in [version 3.4](https://matplotlib.org/3.4.3/gallery/widgets/menu.html) and before:
![grafik](https://github.com/matplotlib/matplotlib/assets/19879328/53e76b55-106e-4004-8299-2c78a03aeaa2)

**Comment by ksunden** [comment]
This does not show up poorly when I run it locally, so not totally sure why it is going wrong on the docs...

**Comment by StefRe** [comment]
Right: `sphx_glr_menu_001.png` is correct and only `sphx_glr_menu_001_2_00x.png` is poorly aligned.

**Comment by anntzer** [comment]
I suspect the "easy" fix would be to reuse the trick of https://matplotlib.org/devdocs/gallery/text_labels_and_annotations/rainbow_text.html to position texts relative to one another.

**Comment by StefRe** [comment]
> I suspect the "easy" fix would be to reuse the trick of https://matplotlib.org/devdocs/gallery/text_labels_and_annotations/rainbow_text.html to position texts relative to one another.

While this solves the alignement issue I couldn't find a way to set all the text boxes to the same length without using `get_window_extent`. That's why I thought it would be easiest to just fake the output, see #27365.

**Comment by jklymak** [comment]
The bbox could be converted to physical units and then all placement done in inches (or points) instead of pixels.  

```
self.text_bbox = self.label.get_window_extent(
            fig.canvas.get_renderer())
self.text_bbox = fig.dpi_scale_trans.inverted().transform_bbox(self.text_bbox)
```

should be in inches.   Then place things using `fig.dpi_scale_trans` rather than native pixels.

**Comment by StefRe** [comment]
Thanks @jklymak, will update the PR.
